tutorials/math/random_number_generation.rst

```diff
@@ -226,6 +414,39 @@ floating-point number between 0.0 and 1.0. We can use this to create a
             # 5% chance of being returned.
             return "Rare"
 
+ .. code-tab:: csharp
+    
+    public override void _Ready()
+    {
+        GD.Randomize();
+
+        foreach (int i in GD.Range(100))
+        {
+            GD.Print(GetItemRarity());
+        }
+    }
+
+    public string GetItemRarity()
+    {
+        var RandomFloat = GD.Randf();
+
+        if (RandomFloat < 0.8f)
+        {
+            // 80% chance of being returned.
+            return "Common";
+        }
+        else if (RandomFloat < 0.95f)
+        {
+            // 15% chance of being returned
+            return "Uncommon";
+        }
+        else
+        {
+            // 5% chance of being returned.
+            return "Rare";
+        }
+    }
+
 .. _doc_random_number_generation_shuffle_bags:
 
 "Better" randomness using shuffle bags
```

Comment:
For what this section is trying to demonstrate, C# doesn't seem to have access to the functions that this covers; `.shuffle()` and `pop_front()` would have to be done manually, but please, correct me if I'm wrong.

Reply:
It'd probably be better to use a `Queue` or a `Stack` for this one, populating it from the original list in a random order.